Add DDA for drawing lines?

loryhndol opened the pull request:
The original function `draw_line` may create some artifacts during rasterization. I would like to add the DDA algorithm because it's simple and easy to understand.

loryhndol committed:
add DDA for drawing lines?

OmarShehata commented:
Hey, thanks for opening this PR!

Do you have a before/after images you can share of what it looks like using this? Also this PR is adding the function but not actually using it to draw, is that intentional?

loryhndol committed:
Update computeRasterizer.wgsl

1. Replace `DDA_round` with builtin function `round` and `max`
2. Describe the procedure with comments

loryhndol commented:
The screenshot below shows the difference between the result of original and my `DDA` implementation. The original result was colored in red and `DDA` result was colored in white. The result was rendered on `<canvas width="539" height="269"></canvas>`.

 I have to admit that the aliasing is quite unnoticeable owing to the high-resolution LCD screen.

![screenshot](https://user-images.githubusercontent.com/36363555/227231194-6b7dbb19-2115-4d2d-9c32-040168f9a8b0.jpg)

Comparison using the `suzanne` model

The original method

![original_suzanne](https://user-images.githubusercontent.com/36363555/227234080-06c7a069-c1b9-4189-b993-798591e0fa74.jpg)

`DDA` method

![DDA_suzanne](https://user-images.githubusercontent.com/36363555/227234067-0e76af83-08e3-4357-a5ff-af40b2a9d864.jpg)

Please see the latest update of the PR. I have made some changes to improve the quality of the code. :)
